python/lsst/meas/extensions/piff/piffPsfDeterminer.py

```diff
@@ -494,12 +532,43 @@ def _get_threshold(nth_order):
                     # No need to do any outlier rejection assume
                     # PSF to be average of few stars.
                     piffConfig['max_iter'] = 1
+                else:
+                    raise PiffTooFewGoodStarsError(
+                        num_good_stars=len(stars),
+                        minimum_dof=threshold,
+                        poly_ndim=piffConfig['interp']['order'],
+                    )
 
         self._piffConfig = piffConfig
         piffResult = piff.PSF.process(piffConfig)
         wcs = {0: gswcs}
 
         piffResult.fit(stars, wcs, pointing, logger=self.piffLogger)
+
+        if piffConfig['interp']['type'] in ['BasisPolynomial', 'Polynomial']:
+            threshold = _get_threshold(piffConfig['interp']['order'])
+            nUsedStars = 0
+            for star in piffResult.stars:
+                if not star.is_flagged and not star.is_reserve:
+                    nUsedStars += 1
+            if nUsedStars <= threshold:
+                if self.config.zerothOrderInterpNotEnoughStars:
+                    self.log.warning(
+                        f"Only {nUsedStars} after outlier rejection, "
+                        f"but {threshold} required. Using zeroth order interpolation."
+                    )
+                    piffConfig['interp']['order'] = 0
+                    # No need to do any outlier rejection assume
+                    # PSF to be average of few stars.
+                    piffConfig['max_iter'] = 1
+                    piffResult.fit(stars, wcs, pointing, logger=self.piffLogger)
+                else:
+                    raise PiffTooFewGoodStarsError(
+                        num_good_stars=nUsedStars,
+                        minimum_dof=threshold,
+                        poly_ndim=piffConfig['interp']['order'],
+                    )
+
         drawSize = 2*np.floor(0.5*stampSize/self.config.samplingSize) + 1
 
         used_image_pos = [s.image_pos for s in piffResult.stars]
```

Comment:
And if check on `is_flagged` and ` not is_reserve` here, we would be propagating out the info of which stars were actually used by setting the `calib_psf_used` flag.